app/id_generation.py

```diff
@@ -2,41 +2,142 @@
 import requests
 from settings import settings
 import random
+from helpers import db_request_token, is_response_valid, is_response_empty
+from request import build_request
+from router import (
+    user as user_router,
+    student as student_router,
+    school,
+    enrollment_record,
+)
 
 school_db_url = settings.db_url + "/school"
 
 
 class JNVIDGeneration:
-    def __init__(self, region, school, grade):
-        self.region = region
-        self.school = school
-        self.grade = grade
-        self.id = (
-            self.get_class_code()
-            + self.get_jnv_code()
-            + self.generate_three_digit_code()
+    def __init__(self, parameters):
+        self.parameters = parameters
+
+        record_already_exist, ID = self.dedupe_for_users()
+        if record_already_exist:
+            self.id = ID
+
+        else:
+            counter = 1000
+            while counter > 0:
+                id = (
+                    self.get_class_code()
+                    + self.get_jnv_code()
+                    + self.generate_three_digit_code()
+                )
+
+                if not self.check_for_duplicate_ID(id):
+                    self.id = id
+                    break
+                counter -= 1
+
+    def dedupe_for_users(self):
```

Comment:
returns a boolean -- whether record exists
and an ID string (empty if record doesn't exist, and id from db if record exists)